src/App.tsx

```diff
@@ -33,171 +117,127 @@ export const App: React.FC = () => {
-              ×
-            </button>
-
-            {/* overlay will cover the todo while it is being deleted or updated */}
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          {/* This todo is an active todo */}
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">
-              Not Completed Todo
+        {filteredTodos.length > 0 && (
+          <section className="todoapp__main" data-cy="TodoList">
+            {filteredTodos.map(({ id, completed, title }) => (
+              <div
+                key={id}
+                data-cy="Todo"
+                className={classNames('todo', { completed })}
+              >
+                <label className="todo__status-label">
+                  <input
+                    data-cy="TodoStatus"
+                    type="checkbox"
+                    className="todo__status"
+                    checked={completed}
+                  />
+                </label>
+                {isEditing[id] ? (
+                  <input
+                    data-cy="TodoTitleField"
+                    type="text"
+                    className="todo__title-field"
+                    placeholder="Empty todo will be deleted"
+                    value={title}
+                  />
+                ) : (
+                  // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+                  <span
+                    data-cy="TodoTitle"
+                    className="todo__title"
+                    onClick={() => toggleEditing(id)}
+                    onKeyDown={event => {
+                      if (event.key === 'Enter') {
+                        toggleEditing(id);
+                      }
+                    }}
+                  >
+                    {title}
+                  </span>
+                )}
+                <button
+                  type="button"
+                  className="todo__remove"
+                  data-cy="TodoDelete"
+                >
+                  ×
+                </button>
+                <div data-cy="TodoLoader" className="modal overlay">
+                  <div className="modal-background has-background-white-ter" />
+                  <div className="loader" />
+                </div>
+              </div>
+            ))}
+          </section>
+        )}
+
+        {shouldShowFooter && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="todosCounter">
```

Comment:
```suggestion
            <span className="todo-count" data-cy="TodosCounter">
```
![image](https://github.com/user-attachments/assets/a790fc3a-c7cf-40e5-8ad0-22fb363b1c03)